android/beagle/src/main/java/br/com/zup/beagle/android/view/mapper/SendRequestActionMapper.kt

```diff
@@ -67,6 +67,14 @@ internal object SendRequestActionMapper {
         return try {
             data.toInt()
         } catch (e: NumberFormatException) {
+            deserializeObjectOrGetRawString(data)
+        } catch (e: Exception) {
+            data
+        }
+    }
+
+    private fun deserializeObjectOrGetRawString(data: String): Any? {
+        return try {
             BeagleMoshi.moshi.adapter(Any::class.java).fromJson(data)
```

Comment:
Here you can use change to:
```suggestion
            BeagleMoshi.moshi.adapter(Any::class.java, ContextDataValue::class.java).fromJson(data)
```

To make sure this Any will be deserialised correctly using the `ContextDataAdapterFactory`.

Reply:
done